### PR TITLE
parametric: golang config remapping

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Add tests for otel config remapping for golang

## Changes

update golang parametric app, regenerate golang protos, fix golang proto generation file script, enable otel config system tests

## Note

don't merge until this PR is released, this PR was tested based on the changes made in the PR linked below
https://github.com/DataDog/dd-trace-go/pull/2748

### CODE REVIEW

File: utils/build/docker/golang/parametric/datadog.go
Comment (mabdinur) [inline utils/build/docker/golang/parametric/datadog.go:108]: @khanayan123 Find a way to access this internal config object

File: tests/parametric/test_otel_env_vars.py
Comment (mabdinur) [inline tests/parametric/test_otel_env_vars.py:181]:
```suggestion
        assert (resp["dd_log_level"] == "debug") or (resp["dd_log_level"] is None
```

File: utils/build/docker/golang/parametric/datadog.go
Comment (mabdinur) [inline utils/build/docker/golang/parametric/datadog.go:153]: We should avoid setting the configuration to null. If the configuration can not be set then we should avoid setting in the config dictionary. Also null isn't idiomatic to golang. Nil would be more appropriate here 

```suggestion
	// config["dd_trace_sample_ignore_parent"] = nil // golang doesn't support DD_TRACE_SAMPLE_IGNORE_PARENT
```

File: manifests/golang.yml
Comment (zacharycmontoya) [inline manifests/golang.yml:417]:
```suggestion
    test_otel_api_interoperability.py: missing_feature
    test_otel_env_vars.py:
      Test_Otel_Env_Vars: v1.66.0
```

File: tests/parametric/test_otel_env_vars.py
Comment (zacharycmontoya) [inline tests/parametric/test_otel_env_vars.py:187]: Maybe we could mark this as `irrelevant` instead of a `missing_feature`?

File: tests/parametric/test_otel_env_vars.py
Comment (zacharycmontoya) [inline tests/parametric/test_otel_env_vars.py:181]: Just like in the context.library check above, I guess that we could refactor this in the following way?

```suggestion
        assert resp["dd_log_level"] in ("debug", None, "null")
```

File: tests/parametric/test_otel_env_vars.py
Comment (zacharycmontoya) [inline tests/parametric/test_otel_env_vars.py:88]:
```suggestion
    @missing_feature(context.library == "golang", reason="DD_LOG_LEVEL is not supported in go")
```

**Review by mabdinur — COMMENTED** [review]
Looks good this is a clever workaround. I recommended some changes to make it a bit resilient to changes

File: manifests/golang.yml
Comment (khanayan123) [reply manifests/golang.yml:417]: i will update this value in this pr once this pr gets merged in:

https://github.com/DataDog/dd-trace-go/pull/2748

File: utils/build/docker/golang/parametric/datadog.go
Comment (mabdinur) [inline utils/build/docker/golang/parametric/datadog.go:124]: Nit: instead of marshaling this json to a map we can use a struct: https://medium.com/@prashant2018/parse-json-string-to-struct-object-in-golang-59d3cf6bce7. This way we can be very intentional about the fields we capture

**Review by mabdinur — COMMENTED** [review]
Left two nits. Overall looks good to me!

**Review by zacharycmontoya — APPROVED** [review]
LGTM

**Review by khanayan123 — APPROVED** [review]

File: tests/parametric/test_otel_env_vars.py
Comment (cbeauchesne) [inline tests/parametric/test_otel_env_vars.py:187]: The test below is skipped with the same reason but with `missing_feature`, is it on purpose ? 

Reminder : 

* `missing_feature` : the feature is not done, but will be done one day. Must be used even if it's not prioritized
* `irrelevant` : the feature can't exists for technical or design choice reasons. This decorator can also be used if the feature won't be implemented for 

The main difference between those two decorator is that system-tests don't  won't bother you with a `XPASS` warning in the CI if one day, a `irrelevant` feature fulfil the test.

**Review by cbeauchesne — APPROVED** [review]
appart from the small detail, AGTM

File: tests/parametric/test_otel_env_vars.py
Comment (mtoffl01) [reply tests/parametric/test_otel_env_vars.py:187]: Given the definitions above + the current implementation/limitations of dd-trace-go, I would consider this feature irrelevant.

File: utils/build/docker/golang/parametric/datadog.go
Comment (mtoffl01) [inline utils/build/docker/golang/parametric/datadog.go:145]: Okay, so you're essentially doing all this logic with reflect in order to iterate over the struct fields and values. Maybe add a comment above line 142 prefacing where you're going, since it wasn't immediately clear to me.

File: utils/build/docker/golang/parametric/datadog.go
Comment (mtoffl01) [inline utils/build/docker/golang/parametric/datadog.go:126]: Can you add a function description above line 126? Since this `Log()` method doesn't just "log," it does a lot more

File: utils/build/docker/golang/parametric/datadog.go
Comment (mtoffl01) [reply utils/build/docker/golang/parametric/datadog.go:150]: Can you also add a comment above like 142 about what this code block is doing/ why you're using reflect (i.e, this is just a path to iterating over the struct fields and values, but that wasn't immediately clear to me at first glance).

File: utils/build/docker/golang/parametric/datadog.go
Comment (mtoffl01) [inline utils/build/docker/golang/parametric/datadog.go:157]:
```suggestion
	// config["dd_log_level"] = nil // dd-trace-go does not support DD_LOG_LEVEL (use DD_TRACE_DEBUG instead)
```

**Comment by mtoffl01** [comment]
It looks like we're testing tracer configuration using the startup log, but it'd be helpful if you could directly orient the reader to that approach using comments

**Review by mtoffl01 — CHANGES_REQUESTED** [review]

File: tests/parametric/test_otel_env_vars.py
Comment (cbeauchesne) [reply tests/parametric/test_otel_env_vars.py:200]: > does not support enabling opentelemetry via DD_TRACE_OTEL_ENABLED

It depends on the *why* you do not support it : 

* it's something you don't support *yet* ? but one day it'll be done, even if you don't know when -> `missing_feature`
* it's something that can't be supported for a technical reason -> definitly `irrelevant`
* it's something you decide to not support for whatever the reason, and you're very unlikely to change your mind -> Good old grey zone :) still `irrelevant`

Comment (mtoffl01) [reply tests/parametric/test_otel_env_vars.py:200]: Yes, `irrelevant` is the right call IMO!

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
`./format.sh` is your friend

**Review by mtoffl01 — APPROVED** [review]

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
A small conflict to fix, plis the CI is failing on the activated test

**Comment by cbeauchesne** [comment]
I set back the PR as draft, you can set it to normal once you need another review

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**
